File: erlang/grains/grains.erl
-module(grains).
-export([square/1, total/0]).

square(N) -> trunc(math:pow(2, N - 1)).

total() -> total(64).

total(Sq) -> total(Sq, 0).

total(0, Acc) -> Acc;
total(N, Acc) -> total(N - 1, Acc + square(N)).

File: clojure/grains/src/grains.clj
(ns grains)

(defn- pow 
  "Raise a number n to exponent e. Uses BigInteger."
  [n e] 
  (.pow (BigInteger. (str n)) e))

(defn square
  "Find the number of grains on a given square."
  [n]
  (pow 2 (- n 1)))

(def square-xf (map square))

(defn total
  "Total number of grains on a chessboard."
  []
  (transduce square-xf + (range 1 65)))
